src/core/ext/filters/client_channel/resolver/polling_resolver.cc

```diff
@@ -121,10 +124,11 @@ void PollingResolver::OnNextResolutionLocked() {
             "[polling resolver %p] re-resolution timer fired: shutdown_=%d",
             this, shutdown_);
   }
-  next_resolution_timer_handle_.reset();
-  if (!shutdown_) {
+  // If we haven't been cancelled nor shutdown, then start resolving.
+  if (next_resolution_timer_handle_.has_value() && !shutdown_) {
     StartResolvingLocked();
   }
+  next_resolution_timer_handle_.reset();
```

Comment:
Suggest moving this up to line 129, since there's no need to reset it if it's already nullopt.

Reply:
Done.